src/components/__tests__/autocomplete.test.js

```diff
@@ -2,28 +2,40 @@ import React from 'react';
 import { ApolloProvider } from '@apollo/client';
 import { fireEvent, render, screen, waitFor } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { client } from '../../ApolloClient';
-import { setupServer } from '../../mocks/setupServer';
+import { CREATE_CLIENT } from '../../ApolloClient';
 import { server } from '../../mocks/server';
 import Autocomplete from '../autocomplete';
 
 const ENTER_KEY_CODE = 13;
 const requestsMap = new Map();
 
+beforeAll(() => {
+  server.listen();
+  server.events.on('request:start', (req) => {
+    requestsMap.set(req.id, req.body.variables);
+  });
+});
+
+afterEach(() => {
+  requestsMap.clear();
+  CREATE_CLIENT.resetStore();
```

Comment:
This would be redundant once the client is instanced per test, not per test suite.